compiler/AST/iterator.cpp

```diff
@@ -412,14 +456,16 @@ buildZip1(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
   }
 
   if (WhileStmt* stmt = toWhileStmt(singleLoop)) {
+    // By the time we get here, the condExpr has been passed through _cond_test
+    // and the result stored in a temp, so condExprForTmpVariableGet just
+    // returns the temp result.  So we can simply say:  (.= this more tmp)
+    // This assumes the backend will implicitly cast tmp to the type of more
```

Comment:
Note to myself so I don't forget to come back to it:

Not sure if this is ok, let me think about it